NtupleProducer/plugins/Ntupler.cc

```diff
@@ -786,16 +788,22 @@ Ntupler::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
             reco::GenParticle genPart = (*genParticles)[iteGen];
             reco::GenParticle genElectron;
             bool fromZ_ele = false;
-            if(abs(genPart.pdgId())==11)
+            if (genPart.isPromptFinalState() && abs(genPart.pdgId())==11 && genPart.fromHardProcessFinalState())
```

Comment:
Can you check the impact of this extra flag ? I think what Shilpi suggested is isPromptFinalState() flag only.